src/lib.rs

```diff
@@ -2575,6 +2575,183 @@ impl<'a> RenderBundleEncoder<'a> {
+    /// }
+    /// ```
+    fn draw_indirect(&mut self, indirect_buffer: &'a Buffer, indirect_offset: BufferAddress);
+
+    /// Draws indexed primitives using the active index buffer and the active vertex buffers,
+    /// based on the contents of the `indirect_buffer`.
+    ///
+    /// The active index buffer can be set with [`RenderBundleEncoder::set_index_buffer`], while the active
+    /// vertex buffers can be set with [`RenderBundleEncoder::set_vertex_buffer`].
+    ///
+    /// The structure expected in `indirect_buffer` is the following:
+    ///
+    /// ```rust
+    /// #[repr(C)]
+    /// struct DrawIndexedIndirect {
+    ///     vertex_count: u32, // The number of vertices to draw.
+    ///     instance_count: u32, // The number of instances to draw.
+    ///     base_index: u32, // The base index within the index buffer.
+    ///     vertex_offset: i32, // The value added to the vertex index before indexing into the vertex buffer.
+    ///     base_instance: u32, // The instance ID of the first instance to draw.
+    /// }
+    /// ```
+    fn draw_indexed_indirect(
+        &mut self,
+        indirect_buffer: &'a Buffer,
+        indirect_offset: BufferAddress,
+    );
+}
+
+impl<'a> RenderEncoder<'a> for RenderPass<'a> {
+    #[inline(always)]
+    fn set_bind_group(&mut self, index: u32, bind_group: &'a BindGroup, offsets: &[DynamicOffset]) {
+        (self as &mut RenderPass<'a>).set_bind_group(index, bind_group, offsets);
+    }
+
+    #[inline(always)]
+    fn set_pipeline(&mut self, pipeline: &'a RenderPipeline) {
+        (self as &mut RenderPass<'a>).set_pipeline(pipeline);
+    }
+
+    #[inline(always)]
+    fn set_index_buffer(&mut self, buffer_slice: BufferSlice<'a>, index_format: IndexFormat) {
+        (self as &mut RenderPass<'a>).set_index_buffer(buffer_slice, index_format);
+    }
+
+    #[inline(always)]
+    fn set_vertex_buffer(&mut self, slot: u32, buffer_slice: BufferSlice<'a>) {
+        (self as &mut RenderPass<'a>).set_vertex_buffer(slot, buffer_slice);
+    }
+
+    #[inline(always)]
+    fn draw(&mut self, vertices: Range<u32>, instances: Range<u32>) {
+        (self as &mut RenderPass<'a>).draw(vertices, instances);
+    }
+
+    #[inline(always)]
+    fn draw_indexed(&mut self, indices: Range<u32>, base_vertex: i32, instances: Range<u32>) {
+        (self as &mut RenderPass<'a>).draw_indexed(indices, base_vertex, instances);
+    }
+
+    #[inline(always)]
+    fn draw_indirect(&mut self, indirect_buffer: &'a Buffer, indirect_offset: BufferAddress) {
+        (self as &mut RenderPass<'a>).draw_indirect(indirect_buffer, indirect_offset);
+    }
+
+    #[inline(always)]
+    fn draw_indexed_indirect(
+        &mut self,
+        indirect_buffer: &'a Buffer,
+        indirect_offset: BufferAddress,
+    ) {
+        (self as &mut RenderPass<'a>).draw_indexed_indirect(indirect_buffer, indirect_offset);
+    }
+}
+
+impl<'a> RenderEncoder<'a> for RenderBundleEncoder<'a> {
```

Comment:
Could we add the features (like push constants) that are in common between renderbundles and renderpasses?

Reply:
I think the only one is `set_push_constants` (which I've added).